stdlib/public/Concurrency/Task.cpp

```diff
@@ -681,7 +690,29 @@ static AsyncTaskAndContext swift_task_create_commonImpl(
       // thread and therefore need to behave a bit more like SC child tasks.
       break;
     }
+    case TaskOptionRecordKind::ResultTypeInfo: {
+      auto *typeInfo = cast<ResultTypeInfoTaskOptionRecord>(option);
+#if SWIFT_CONCURRENCY_EMBEDDED
```

Comment:
Should this whole case be embedded-only, or should we assert if we hit this case on non-embedded?

Reply:
I need the enum case to be visible to the compiler (it's used in IRGen), so I went with the assert instead.